harness/go/harness/util.go

```diff
@@ -41,6 +42,72 @@ func FindEvent(history client.HistoryEventIterator, cond func(*historypb.History
 	return nil, nil
 }
 
+// ExecuteWithArgs runs a workflow with default arguments
+func ExecuteWithArgs(workflow interface{}, args ...interface{}) func(ctx context.Context, r *Runner) (client.WorkflowRun, error) {
+	return func(ctx context.Context, r *Runner) (client.WorkflowRun, error) {
+		opts := r.Feature.StartWorkflowOptions
+		if opts.TaskQueue == "" {
+			opts.TaskQueue = r.TaskQueue
+		}
+		if opts.WorkflowExecutionTimeout == 0 {
+			opts.WorkflowExecutionTimeout = 1 * time.Minute
+		}
+		return r.Client.ExecuteWorkflow(ctx, opts, workflow, args...)
+	}
+}
+
+// GetWorkflowResultPayload returns the first payload of a workflow result
+func GetWorkflowResultPayload(
+	ctx context.Context,
+	client client.Client,
+	workflowID string,
+) (*commonpb.Payload, error) {
+	history := client.GetWorkflowHistory(ctx, workflowID, "", false, enumspb.HISTORY_EVENT_FILTER_TYPE_CLOSE_EVENT)
+
+	event, err := FindEvent(history, func(ev *historypb.HistoryEvent) bool {
+		attrs := ev.GetWorkflowExecutionCompletedEventAttributes()
+		return attrs != nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	if event == nil {
+		return nil, errors.New("missing WorkflowExecutionCompleted event")
+	}
+
+	attrs := event.GetWorkflowExecutionCompletedEventAttributes()
+	if attrs == nil {
+		return nil, errors.New("could not locate WorkflowExecutionCompleted event")
+	}
+	return attrs.GetResult().GetPayloads()[0], nil
```

Comment:
You can use the fields when you are not concerned about nil safety (e.g. `attrs.Result.Payloads[0]`), but you may want to check for nil result or empty payloads. No big deal though.